include/zephyr/drivers/emul.h

```diff
@@ -18,9 +18,23 @@
 extern "C" {
 #endif /* __cplusplus */
 
-struct device;
 struct emul;
```

Comment:
@yperess 

The includes must come after the forward-declaration otherwise we get their usages before the type's been declared when we include the other headers, leading to a *real* compilation failure.